crates/env/src/engine/on_chain/impls.rs

```diff
@@ -395,6 +396,7 @@ impl TypedEnvBackend for EnvInstance {
         // 1024 bytes. Beyond that limit ink! contracts will trap for now.
         // In the default configuration encoded `AccountId` require 32 bytes.
         let out_address = &mut scoped.take(1024);
+        let salt = scoped.take_bytes(params.salt_bytes().as_ref());
```

Comment:
With `Salt: AsRef<[u8]>` this is not needed and would just cause copy overhead. Simply take the `salt_bytes().as_ref()` directly.